src/server/scripts/Kalimdor/zone_felwood.cpp

```diff
@@ -28,10 +28,94 @@ EndScriptData */
 EndContentData */
 
 #include "ScriptMgr.h"
+#include "script_helper.h"
 #include "ScriptedCreature.h"
 #include "ScriptedGossip.h"
 #include "Player.h"
 
+class npc_impsy_47365 : public CreatureScript
+{
+public:
+    npc_impsy_47365() : CreatureScript("npc_impsy_47365") {}
+
+    enum script_enums
+    {
+        ENCHANTED_IMP_SAC = 88354,
+        QUEST_CREDIT = 47365
+    };
+
+    struct npc_impsy_47365AI : public ScriptedAI
+    {
+        npc_impsy_47365AI(Creature* creature) : ScriptedAI(creature) { Initialize(); }
+
+        EventMap  _events;
+        Position  _spawn_position;
+
+        void Initialize()
+        {
+        }
+
+        void Reset() override
+        {
+            float x, y, z;
+            me->GetRespawnPosition(x, y, z);
+            _spawn_position = Position(x, y, z);
+        }
+
+        void SpellHit(Unit* caster, SpellInfo const* spell) override
+        {
+            if (Player* player = caster->ToPlayer())
+                if (spell->Id == ENCHANTED_IMP_SAC)
+                {
+                    player->KilledMonsterCredit(QUEST_CREDIT);
+                    me->DespawnOrUnsummon(0);
+                }
+        }
+
+        void EnterCombat(Unit* who) override
+        {
+            _events.ScheduleEvent(EVENT_CHECK_FOR_PLAYER, 1000);
+        }
+
+
+        void UpdateAI(uint32 diff) override
+        {
+            _events.Update(diff);
+
+            while (uint32 eventId = _events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                case EVENT_CHECK_FOR_PLAYER:
+                {
+                    if (me->IsAlive())
+                    {
+                        if (me->GetDistance(_spawn_position) > 50)
+                        {
+                            me->CombatStop();
+                            me->GetMotionMaster()->MovePoint(0, _spawn_position);
+                        }
+                        else
+                        {
+                            me->AI()->DoMeleeAttackIfReady();
+                            _events.ScheduleEvent(EVENT_CHECK_FOR_PLAYER, 1000);
+                        }
+                    }
+
+                }
+                }
+            }
+        }
+
+    };
+
+    CreatureAI* GetAI(Creature* creature) const
+    {
+        return new npc_impsy_47365AI(creature);
+    }
+};
+
 void AddSC_felwood()
 {
+    new npc_impsy_47365();
 }
```

Comment:
look at video: https://www.youtube.com/watch?v=fk0x_E3GdFk player has to remove all armor, weapon and cloth.. you have to check it befor cast spell

Reply:
Ok i will watch the video.